src/filter/fast_blur.rs

```diff
@@ -0,0 +1,163 @@
+use std::cmp::{max, min};
+
+use image::{ImageBuffer, Pixel, Primitive};
+
+/// Approximation of Gaussian blur after
+/// Kovesi, P.:  Fast Almost-Gaussian Filtering The Australian Pattern
+/// Recognition Society Conference: DICTA 2010. December 2010. Sydney.
+pub fn fast_blur<P: Pixel>(
+    image_buffer: &ImageBuffer<P, Vec<P::Subpixel>>,
+    sigma: f32,
+) -> ImageBuffer<P, Vec<P::Subpixel>> {
+    let (width, height) = image_buffer.dimensions();
+    let mut samples = image_buffer.as_flat_samples().samples.to_vec();
+    let num_passes = 3;
+
+    let boxes = boxes_for_gauss(sigma, num_passes);
+
+    for radius in boxes.iter().take(num_passes) {
+        let horizontally_blurred_transposed = my_fast_horizontal_blur::<P::Subpixel>(
+            &samples,
+            width as usize,
+            height as usize,
+            *radius,
+            P::CHANNEL_COUNT as usize,
+        );
+        samples = my_fast_horizontal_blur::<P::Subpixel>(
+            &horizontally_blurred_transposed,
+            height as usize,
+            width as usize,
+            *radius,
+            P::CHANNEL_COUNT as usize,
+        );
+    }
+    ImageBuffer::from_raw(width, height, samples).unwrap()
+}
+
+fn boxes_for_gauss(sigma: f32, n: usize) -> Vec<usize> {
+    let w_ideal = f32::sqrt((12.0 * sigma * sigma / (n as f32)) + 1.0);
+    let mut w_l = w_ideal.floor();
+    if w_l % 2.0 == 0.0 {
+        w_l -= 1.0
+    };
+    let w_u = w_l + 2.0;
+
+    let m_ideal = (12.0 * sigma * sigma
```

Comment:
Expression is too long, I'm sure there is a way to simplify it

Reply:
It`s II (5) in https://peterkovesi.com/papers/FastGaussianSmoothing.pdf. It results from inverting the equation of ideal sigma and the one coming from repeated box blurs. I don't know how to further simplify this polynomial as all of the terms are linearly independent (as polynomials).

Reply:
```rs
let m_ideal = 0.25 * (n as f32) * (w_l + 3.0) - 3.0 * sigma.powi(2) * (w_l + 1.0).recip();
```

Reply:
Thank you, now the formula fits into one line.